exporters/otlp/src/otlp_grpc_exporter.cc

```diff
@@ -52,37 +60,85 @@ sdk::common::ExportResult OtlpGrpcExporter::Export(
+  proto::collector::trace::v1::ExportTraceServiceRequest *request =
+      google::protobuf::Arena::Create<proto::collector::trace::v1::ExportTraceServiceRequest>(
+          arena.get());
+  OtlpRecordableUtils::PopulateRequest(spans, request);
 
-  grpc::Status status =
-      OtlpGrpcClient::DelegateExport(trace_service_stub_.get(), context.get(), request, &response);
+  auto context = OtlpGrpcClient::MakeClientContext(options_);
+  proto::collector::trace::v1::ExportTraceServiceResponse *response =
+      google::protobuf::Arena::Create<proto::collector::trace::v1::ExportTraceServiceResponse>(
+          arena.get());
 
-  if (!status.ok())
+#ifdef ENABLE_ASYNC_EXPORT
+  if (options_.max_concurrent_requests > 1)
   {
-    OTEL_INTERNAL_LOG_ERROR("[OTLP TRACE GRPC Exporter] Export() failed with status_code: \""
-                            << grpc_utils::grpc_status_code_to_string(status.error_code())
-                            << "\" error_message: \"" << status.error_message() << "\"");
-    return sdk::common::ExportResult::kFailure;
+    return client_->DelegateAsyncExport(
+        options_, trace_service_stub_.get(), std::move(context), std::move(arena),
+        std::move(*request),
+        [](opentelemetry::sdk::common::ExportResult result,
+           std::unique_ptr<google::protobuf::Arena> &&,
+           const proto::collector::trace::v1::ExportTraceServiceRequest &request,
+           proto::collector::trace::v1::ExportTraceServiceResponse *) {
+          if (result != opentelemetry::sdk::common::ExportResult::kSuccess)
+          {
+            OTEL_INTERNAL_LOG_ERROR("[OTLP HTTP Client] ERROR: Export "
+                                    << request.resource_spans_size()
+                                    << " trace span(s) error: " << static_cast<int>(result));
+          }
+          else
+          {
+            OTEL_INTERNAL_LOG_DEBUG("[OTLP HTTP Client] Export " << request.resource_spans_size()
+                                                                 << " trace span(s) success");
+          }
+          return true;
+        });
+  }
+  else
+  {
+#endif
+    grpc::Status status =
+        OtlpGrpcClient::DelegateExport(trace_service_stub_.get(), std::move(context),
+                                       std::move(arena), std::move(*request), response);
+    if (!status.ok())
+    {
+      OTEL_INTERNAL_LOG_ERROR("[OTLP TRACE GRPC Exporter] Export() failed with status_code: \""
+                              << grpc_utils::grpc_status_code_to_string(status.error_code())
+                              << "\" error_message: \"" << status.error_message() << "\"");
+      return sdk::common::ExportResult::kFailure;
+    }
+#ifdef ENABLE_ASYNC_EXPORT
   }
+#endif
   return sdk::common::ExportResult::kSuccess;
 }
 
-bool OtlpGrpcExporter::ForceFlush(std::chrono::microseconds /* timeout */) noexcept
+bool OtlpGrpcExporter::ForceFlush(std::chrono::microseconds timeout) noexcept
```

Comment:
Probably needs some MAYBE_UNUSED to avoid compile warnings without async.
Same with every force flush / shutdown method.

Reply:
Thanks and one.